### ISSUE TITLE
How to control the training order between disp_net and pose_net

### ISSUE THREAD

**Issue by royal-feng** [drmaj/UnDeepVO#1]
Surprise to see the Implementation of UnDeepVO, I also do the implementation work recently. But I wonder how the training order between disp_net and pose_net? Because we use the depth_net to reconstruct the real depth ,then use the depth to estimate the pose between the adjacent frame. So I wonder maybe train the depth_net for several epoch(such 30 epoch),then training the pose_net for next epoch. I wonder in the early training, the depth net is not so dependable. Could you give me some advice for how to consider this?

**Comment by ghost** [comment]
@royal-feng Although I have not finished the implementation, it seems to me that [Li et. al](https://arxiv.org/abs/1709.06841) trains both the depth map net and pose net simultaneously. 

> 

> By using these constraints to construct the loss function and minimizing them all together, the UnDeepVO learns to estimate scaled 6-DoF poses and depth maps in an end-to- end unsupervised manner.

pg. 2 , section II, last sentence.

As you mentioned, you could try to train the depth map net first until you have a decent level of convergence and then use it to compute the losses while training the pose net (since both temporal loss functions rely on depth). I am trying to replicate the paper directly as best possible, so I will try to train both nets simultaneously, perhaps using auxiliary lambda layers or custom loss functions.

**Comment by ghost** [comment]
Closed.
